crates/forge/bin/cmd/script/multi.rs

```diff
@@ -139,37 +139,35 @@ impl ScriptArgs {
                 join_all(futs).await.into_iter().filter(|res| res.is_err()).collect::<Vec<_>>();
 
             if !errors.is_empty() {
-                return Err(eyre::eyre!("{errors:?}"))
+                return Err(eyre::eyre!("{errors:?}"));
             }
         }
 
         trace!(target: "script", "broadcasting multi chain deployments");
 
-        let futs = deployments
-            .deployments
-            .iter_mut()
-            .map(|sequence| async {
-                match self
-                    .send_transactions(
-                        sequence,
-                        &sequence.typed_transactions().first().unwrap().0.clone(),
-                        &script_wallets,
-                    )
-                    .await
-                {
-                    Ok(_) => {
-                        if self.verify {
-                            return sequence.verify_contracts(config, verify.clone()).await
-                        }
-                        Ok(())
+        let mut results: Vec<Result<(), Report>> = Vec::new(); // Declare a Vec of Result<(), Report>
+
+        for sequence in deployments.deployments.iter_mut() {
+            let result = match self
+                .send_transactions(
+                    sequence,
+                    &sequence.typed_transactions().first().unwrap().0.clone(),
+                    &script_wallets,
+                )
+                .await
+            {
+                Ok(_) => {
+                    if self.verify {
+                        return sequence.verify_contracts(config, verify.clone()).await
                     }
-                    Err(err) => Err(err),
+                    Ok(())
                 }
-            })
-            .collect::<Vec<_>>();
+                Err(err) => Err(err),
+            };
+            results.push(result);
+        }
 
-        let errors =
-            join_all(futs).await.into_iter().filter(|res| res.is_err()).collect::<Vec<_>>();
+        let errors = results.into_iter().filter(|res| res.is_err()).collect::<Vec<_>>();
```

Comment:
Right, so I see what's going on:
- Before, we drove all futures to completion depending on availability (ideally in order), but this wasn't guaranteed—what's guaranteed is that they're stored in the order they were provided.
- Now, we're waiting for each batch to complete and storing its result, avoiding the race condition

Reply:
yeah, but the potential temporal dependencies between the batches would result in script failures. 

I could still see people running issues if their scripts use blockchain state that changes too quickly even after one block passing: i.e. getting the price of ETH in a ETH/USDC uni pool, and using that as an input to another script.

But for the most part I feel like people just wanna deploy their contracts & maybe do some configurations on them. Not so much use specific chain-state.

Reply:
and yeah, complex multichain deployments would take longer. I feel like it's a worthwhile tradeoff for the amount of flexibility that's enabled by it.